### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by Benjamin-Freoua-Alma**
### Reason for change

Change the return code (500 to 200)on Ipn callback when it's a mismatch or fraud

[Change the return code](https://linear.app/almapay/issue/MPP-410/refund-mismatch-need-return-200-in-ipn)
[sepa payment](https://linear.app/almapay/issue/MPP-423/fix-issue-sepa)
[improve purchaseAmount verification](https://linear.app/almapay/issue/MPP-437/improve-purchaseamount-verification)

### How to test

_As a reviewer, you are encouraged to test the PR locally._

Make a mismatch and test with IPN

### Checklist for authors and reviewers

- [ ] The title of the PR uses business wording, not technical jargon, for the changelog readers to understand it
- [ ] The PR implements the changes asked in the referenced task / issue
- [ ] You understand the impact of this PR on existing code/features

### Non applicable

- [ ] The automated tests are compliant with the [testing strategy](https://www.notion.so/almapay/Backend-testing-strategy-06c642cec1bf47b9b8feca3a91ea8d4a)
- [ ] The tests are relevant, and cover the corner/error cases, not only the happy path
- [ ] The changes include adequate logging and Datadog traces
- [ ] Documentation is updated (API, developer documentation, ADR, Notion...)

### CODE REVIEW

**Review by joyet-simon — APPROVED** [review]

File: alma/lib/API/PaymentValidation.php [hunk 1]
```diff
@@ -161,13 +169,14 @@ public function validatePayment($almaPaymentId)
...
-                $refundHelper = new RefundHelper($this->module, $cart, $payment->id);
+                $clientHelper = new ClientHelper();
+                $refundHelper = new RefundHelper($this->module, $cart, $payment->id, $clientHelper);
                 $refundHelper->mismatchFullRefund();
             }
 
             $firstInstalment = $payment->payment_plan[0];
             if (!in_array($payment->state, [Payment::STATE_IN_PROGRESS, Payment::STATE_PAID])
-                || $firstInstalment->state !== Instalment::STATE_PAID
+                || Instalment::STATE_PAID !== $firstInstalment->state
```
Comment (Francois-Gomis) [inline alma/lib/API/PaymentValidation.php:179]: You must remove test on firstInstalment->state for SEPA payment

**Comment by ClaireAlma** [comment]
I don't understand the description of this PR

File: alma/lib/API/MismatchException.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+class MismatchException extends Exception
+{
+    /**
+     * Construct Mismatch Exception
+     *
+     * @param $message
+     * @param $code
+     * @param $previous
+     */
+    public function __construct($message, $code = 0, $previous = null)
```
Comment (ClaireAlma) [inline alma/lib/API/MismatchException.php:40]: if you don't extend the construct, don't put it

**Review by joyet-simon — APPROVED** [review]

**Review by Francois-Gomis — APPROVED** [review]

**Review by joyet-simon — APPROVED** [review]

**Comment by ClaireAlma** [comment]
Just edit the changelog and put the Sepa ticket in description

**Review by ClaireAlma — CHANGES_REQUESTED** [review]
Just edit the changelog and put the Sepa ticket in description

**Review by ClaireAlma — APPROVED** [review]

**Pull request merged**